### PR TITLE
[Open] Update Spec

### PR DESCRIPTION
**Pull request by Monkeyank**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature
**What is the current behavior? (You can also link to an open issue here)**
Currently open doesn't have any options, but just suggests filepaths/directories.
**What is the new behavior (if this is a feature change)?**
Adds options!
**Additional info:**

### CODE REVIEW

File: dev/open.ts [hunk 1]
```diff
@@ -1,7 +1,82 @@
...
+    {
+      name: "-h",
+      description:
+        "Searches header locations for a header whose name matches the given string and then opens it",
+    },
+    {
+      name: "--args",
+      description:
+        "All remaining arguments are passed to the opened application in the argv parameter to main().  These arguments are not opened or interpreted by the open tool",
+    },
```
Comment (mattschrage) [inline dev/open.ts:76]: I think this should have a variadic `arg` object.

**Review by mattschrage — CHANGES_REQUESTED** [review]
I'm excited for this one! (Especially if you can get the icon API working for the applications - will be very useful and cool)

File: dev/open.ts [hunk 2]
```diff
@@ -1,7 +1,99 @@
...
+          },
+          {
+            script: "ls -1p ~/Applications",
+            postProcess: function (out) {
+              return out.split("\n").map((line) => ({
+                name: line,
+                icon: `fig://~/Applications/${line}`,
+                priority: line.endsWith(".app/") && 76,
+              }));
+            },
```
Comment (QuiiBz) [inline dev/open.ts:29]: Those generators are amazing! But there is a bit of duplication, let's create a top-level function to generate a generator:

```ts
const appsGenerator = (path: string): Fig.Generator => ({
  script: `ls -1p ${path}`,
  postProcess: (out) => {
    return out.split("\n").map((line) => ({
      name: line,
      icon: `fig://${path}/${line}`,
      priority: line.endsWith(".app/") && 76,
    }));
  },
});
```

And we can use it like this:
```suggestion
          appsGenerator("/Applications"),
          appsGenerator("~/Applications"),
```

Comment (Monkeyank) [reply dev/open.ts:29]: Haha, just was about to push this up 😅

**Review by QuiiBz — APPROVED** [review]
Amazing job! Love the generators with the icons

**Pull request merged**